Add Tabs to MainPanel to show both Owned and Shared Worksheets

yuqijin opened the pull request:
### Reasons for making this change

- Fetch only the first 100 worksheets to avoid overload
- Add a tooltip to remind user only the first 100 worksheets are shown
- Add Tabs to the main panel to show both "Owned Worksheets" and "Shared Worksheets"
- Shared Worksheets Tab is only enabled for the authorized user

### Related issues

fixes #3408 

### Screenshots

<img width="1640" alt="Screen Shot 2021-04-22 at 1 00 22 PM" src="https://user-images.githubusercontent.com/34461466/115778118-01165100-a36b-11eb-9267-22b23bc6922b.png">

![ezgif com-gif-maker (3)](https://user-images.githubusercontent.com/34461466/115777665-6ddd1b80-a36a-11eb-93b1-b834b108265e.gif)


### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

Commit:
Add Tabs to MainPanel to show both "Owned Worksheets" and "Shared Worksheets"

Commit:
Adjust margin for ADD button

epicfaace reviewed (commented):
Looks good!
- Can we rename "Owned Worksheets" to "My Worksheets"?
- For the 100-worksheet message, instead of showing it as a tooltip, perhaps it would be best to just show it as some text at the bottom of the page, once the user scrolls down to see all the worksheets? I think it would only be relevant once the user scrolls down to the bottom.

yuqijin commented:
> Looks good!
> 
> * Can we rename "Owned Worksheets" to "My Worksheets"?
> * For the 100-worksheet message, instead of showing it as a tooltip, perhaps it would be best to just show it as some text at the bottom of the page, once the user scrolls down to see all the worksheets? I think it would only be relevant once the user scrolls down to the bottom.

- Updated according to today's meeting's discussion.
- I think your suggestion for the 100-worksheet msg is meaningful! I have already removed the tooltip and move the message at the bottom of the page.
<img width="1680" alt="Screen Shot 2021-04-22 at 10 10 09 PM" src="https://user-images.githubusercontent.com/34461466/115821654-077eea00-a3b8-11eb-88e3-cbc17b5447e3.png">

My Profile:
<img width="1239" alt="Screen Shot 2021-04-23 at 2 40 51 PM" src="https://user-images.githubusercontent.com/34461466/115946010-563d8a00-a473-11eb-90ec-a9712542d727.png">

Other user's Profile:
<img width="1238" alt="Screen Shot 2021-04-23 at 2 41 05 PM" src="https://user-images.githubusercontent.com/34461466/115946014-589fe400-a473-11eb-9b8a-052ef3c64498.png">

Commit:
Show the 100-worksheet message at the bottom of the page

Commit:
Improve tab name

yuqijin committed:
Merge branch 'master' into fix/3408-new-dashboard

yuqijin committed:
Update frontend/src/components/Dashboard/MainPanel.js

Co-authored-by: Ashwin Ramaswami <aramaswamis@gmail.com>

mergify[bot] committed:
Merge branch 'master' into fix/3408-new-dashboard

mergify[bot] committed:
Merge branch 'master' into fix/3408-new-dashboard

mergify[bot] committed:
Merge branch 'master' into fix/3408-new-dashboard
